ibazel/main.go

```diff
@@ -94,27 +101,54 @@ func main() {
 		return
 	}
 
-	// Listen to the events and trigger action based on the response code.
-	go func() {
-		for {
+	state := QUERY
+	for {
+		fmt.Printf("State: %s\n", state)
+		switch state {
+		case WAIT:
+			select {
+			case <-sourceFileWatcher.Events:
+				fmt.Printf("Detected source change. Rebuilding...\n")
+				state = DEBOUNCE_RUN
+			case <-buildFileWatcher.Events:
+				fmt.Printf("Detected build graph change. Requerying...\n")
+				state = DEBOUNCE_QUERY
+			}
+		case DEBOUNCE_QUERY:
+			state = QUERY
+			select {
```

Comment:
would this be more readable as a:

```
for {
  select {
  case <-time.After(debounceDuration):
    break
  case <-buildFileWatcher.Events:
    continue
  }
}
```

I find it a bit confusing that it switches back and forth between query and debounce here, and continues the outer loop.